src/vmm/src/vmm_config/machine_config.rs

```diff
@@ -217,27 +255,9 @@ where
     Ok(val)
 }
 
-/// Deserialization function for the `cpu_template` field in `VmConfig` and `VmUpdateConfig`.
-/// This is called only when `cpu_template` is present in the JSON configuration.
-fn deserialize_cpu_template<'de, D, T>(_d: D) -> std::result::Result<T, D::Error>
-where
-    D: de::Deserializer<'de>,
-    T: Deserialize<'de>,
-{
-    // If this function was called it means that `cpu_template` was specified in
-    // the JSON. Return an error since `cpu_template` is not supported on aarch64.
-    #[cfg(target_arch = "aarch64")]
-    return Err(de::Error::invalid_value(
-        de::Unexpected::Enum,
-        &"CPU templates are not supported on aarch64",
-    ));
-
-    #[cfg(target_arch = "x86_64")]
-    T::deserialize(_d)
-}
-
-/// Template types available for configuring the CPU features that map
+/// Template types available for configuring the x86 CPU features that map
 /// to EC2 instances.
+#[cfg(target_arch = "x86_64")]
```

Comment:
If we create a `CpuFeaturesTemplate` set aside specifically for each architecture, we should move the enum into their sub-modules IMO.

Reply:
done